lldb/tools/lldb-dap/DAPError.h

```diff
@@ -0,0 +1,33 @@
+//===-- DAPError.h --------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Support/Error.h"
+#include <string>
+
+namespace lldb_dap {
+
+/// An Error that is reported as a DAP Error Message, which may be presented to
+/// the user.
+class DAPError : public llvm::ErrorInfo<DAPError> {
+public:
+  static char ID;
+
+  DAPError(std::string message, bool show_user = false);
```

Comment:
Why the default to false for show_user? I can't think of an example of an error that I wouldn't want to surface (off the top of my head).

Additionally, I think this should probably not have a default and every construction should set it explicitly, but again I can't think of a scenario where I wouldn't want to show it

Reply:
I adjusted this to default to `true`. The field is optional in [Message](https://microsoft.github.io/debug-adapter-protocol/specification#Types_Message) but I think your right that most of the time we're creating a DAPError we'd want to show the user.

For reference, when an error is shown, it looks like:
<img width="476" alt="Screenshot 2025-03-20 at 12 03 03 PM" src="https://github.com/user-attachments/assets/8c2582d6-b25f-4973-b282-1c6297af9d9e" />

Reply:
I updated the DAPError to allow callers to set a few additional fields as well. I'm not sure if we'll use them all yet, but they're there if we want to add a URL link to an error message.

Reply:
I think this is great!